lib/ml_egl.c

```diff
@@ -411,6 +491,85 @@ void ml_egl_set_motion_notify_callback(value v)
   if(!initialized)
     failwith("Egl.set_motion_notify_callback: not initialized") ;
   caml_modify_generational_global_root(&motion_notify_callback, v) ;
+  XEventMask = XEventMask | PointerMotionMask ;
+  XSelectInput(xdisplay, xwindow, XEventMask);
+  CAMLreturn0 ;
+}
+
+void ml_egl_unset_motion_notify_callback()
+{
+  CAMLparam0() ;
+  if(!initialized)
+    failwith("Egl.unset_motion_notify_callback: not initialized") ;
+  caml_modify_generational_global_root(&motion_notify_callback, default_callback) ;
+  XEventMask = XEventMask & ~PointerMotionMask ;
+  XSelectInput(xdisplay, xwindow, XEventMask);
+  CAMLreturn0 ;
+}
+
+void ml_egl_set_enter_window_callback(value v)
+{
+  CAMLparam1(v) ;
+  if(!initialized)
+    failwith("Egl.set_motion_notify_callback: not initialized") ;
+  caml_modify_generational_global_root(&enter_window_callback, v) ;
+  XEventMask = XEventMask | EnterWindowMask ;
+  XSelectInput(xdisplay, xwindow, XEventMask);
+  CAMLreturn0 ;
+}
+
+void ml_egl_unset_enter_window_callback()
+{
+  CAMLparam0() ;
+  if(!initialized)
+    failwith("Egl.unset_motion_notify_callback: not initialized") ;
+  caml_modify_generational_global_root(&enter_window_callback, default_callback) ;
+  XEventMask = XEventMask & ~EnterWindowMask ;
+  XSelectInput(xdisplay, xwindow, XEventMask);
+  CAMLreturn0 ;
+}
+
+void ml_egl_set_leave_window_callback(value v)
+{
+  CAMLparam1(v) ;
+  if(!initialized)
+    failwith("Egl.set_motion_notify_callback: not initialized") ;
```

Comment:
I noticed the error message is wrong here, we should check them all before merging.

Reply:
By the way, these functions are all very similar. Maybe we should generate them with a macro.